internal/planner/server.go

```diff
@@ -92,3 +99,36 @@ func (s *Server) createFlavor(w http.ResponseWriter, r *http.Request) {
 
 	http.Error(w, fmt.Sprintf("failed to create flavor: %v", err), http.StatusInternalServerError)
 }
+
+// getFlavorPressure handles retrieving the pressure for a specific or all flavors.
+func (s *Server) getFlavorPressure(w http.ResponseWriter, r *http.Request) {
+	var pressures = map[string]int{}
+	var err error
+
+	flavorName := r.PathValue("name")
+	switch flavorName {
+	case allFlavorName:
+		pressures, err = s.store.GetAllPressures(r.Context(), flavorPlatform)
+	default:
+		pressures, err = s.store.GetPressures(r.Context(), flavorPlatform, flavorName)
+	}
+
+	if err == nil {
+		respondWithJSON(w, http.StatusOK, pressures)
+		return
+	}
+
+	if errors.Is(err, database.ErrNotExist) {
```

Comment:
right now the database won't return `ErrNotExist`, we can just replace this by checking if the pressures returned by the database is actually empty